test/core/table_fill.wast

```diff
@@ -0,0 +1,154 @@
+(module
+  (table $t 10 anyref)
+
+  (func (export "fill") (param $i i32) (param $r anyref) (param $n i32)
+    (table.fill $t (local.get $i) (local.get $r) (local.get $n))
+  )
+
+  (func (export "get") (param $i i32) (result anyref)
+    (table.get $t (local.get $i))
+  )
+)
+
+(assert_return (invoke "get" (i32.const 1)) (ref.null))
+(assert_return (invoke "get" (i32.const 2)) (ref.null))
+(assert_return (invoke "get" (i32.const 3)) (ref.null))
+(assert_return (invoke "get" (i32.const 4)) (ref.null))
+(assert_return (invoke "get" (i32.const 5)) (ref.null))
+
+(assert_return (invoke "fill" (i32.const 2) (ref.host 1) (i32.const 3)))
+(assert_return (invoke "get" (i32.const 1)) (ref.null))
+(assert_return (invoke "get" (i32.const 2)) (ref.host 1))
+(assert_return (invoke "get" (i32.const 3)) (ref.host 1))
+(assert_return (invoke "get" (i32.const 4)) (ref.host 1))
+(assert_return (invoke "get" (i32.const 5)) (ref.null))
+
+(assert_return (invoke "fill" (i32.const 4) (ref.host 2) (i32.const 2)))
+(assert_return (invoke "get" (i32.const 3)) (ref.host 1))
+(assert_return (invoke "get" (i32.const 4)) (ref.host 2))
+(assert_return (invoke "get" (i32.const 5)) (ref.host 2))
+(assert_return (invoke "get" (i32.const 6)) (ref.null))
+
+(assert_return (invoke "fill" (i32.const 4) (ref.host 3) (i32.const 0)))
+(assert_return (invoke "get" (i32.const 3)) (ref.host 1))
+(assert_return (invoke "get" (i32.const 4)) (ref.host 2))
+(assert_return (invoke "get" (i32.const 5)) (ref.host 2))
+
+(assert_return (invoke "fill" (i32.const 8) (ref.host 4) (i32.const 2)))
+(assert_return (invoke "get" (i32.const 7)) (ref.null))
+(assert_return (invoke "get" (i32.const 8)) (ref.host 4))
+(assert_return (invoke "get" (i32.const 9)) (ref.host 4))
+
+(assert_return (invoke "fill" (i32.const 9) (ref.null) (i32.const 1)))
+(assert_return (invoke "get" (i32.const 8)) (ref.host 4))
+(assert_return (invoke "get" (i32.const 9)) (ref.null))
+
+(assert_return (invoke "fill" (i32.const 10) (ref.host 5) (i32.const 0)))
+(assert_return (invoke "get" (i32.const 9)) (ref.null))
+
+(assert_trap
+  (invoke "fill" (i32.const 8) (ref.host 6) (i32.const 3))
+  "out of bounds"
```

Comment:
In this case we also want to check that there was partial initialization of the elements up to the bound.

Reply:
Isn't that what the next 3 asserts are doing, or did you mean something else?

Reply:
Sorry, my oversight - eyes glazing over...